keps/sig-node/4569-cgroup-v1-maintenance-mode/README.md

```diff
@@ -0,0 +1,641 @@
+
+### API Changes
+
+N/A
+
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+The goal is to ensure that we don't accept enhancements with inadequate testing.
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[X] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+All existing test jobs that use cgroup v2 should continue to pass without any flakiness.
+
+##### Unit tests
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+The respective kubelet subcomponents already have unit tests cases to handle cgroup v2.
+
+##### Integration tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+N/A
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+1. Monitor existing cgroup v2 jobs, e.g.
+  - https://testgrid.k8s.io/sig-node-release-blocking#ci-crio-cgroupv2-node-e2e-conformance
+
+2. Ensure all features coverage by running all tests on cgroup v2 (while some may still run on cgroup v1 to test back compatibility)
+
+### Graduation Criteria
```

Comment:
Hmm.... I'm a bit perplexed about the alpha/beta/ga stages listed here. Based on the earlier section, in 1.31, we'll log a warning and send an event about cgroup v1 been transitioned into maintenance mode. Then do we really need to go through all three stages? We can just merged them into one.

Reply:
The purpose of the `alpha` phase is to publicly announce that cgroup v1 is transitioning into maintenance mode. This early warning will help users prepare for the migration to the newer version.

During the `beta` phase, we can dedicate time to ensure that CI systems align properly, that the cgroup manager add both v1 and v2 specific implementations, and that we address any known issues with v2.

Finally, the `GA` phase is when we officially update the Kubernetes documentation to reflect that cgroup v1 is now in maintenance mode.

Reply:
Hmm... I have a different opinion. Putting cgroupv1 into maintenance mode has impacts on developers and end users. 

1. For end user, the real impact is the "best-effort bug fixes" mentioned in the KEP. But the rest doesn't affect the them yet as we don't have a timeline to stop supporting cgroupv2 yet.
2. For developers, the feature-freeze for cgroup v1 is the primary thing. In reality, there are features that already don't support cgroupv1 by design.

Can we accelerate (2) by declaring feature freeze for cgroupv1 as early as possible (e.g., in the alpha stage)? This lowers the development/support burden and matches the reality better. 
For (1), we can delay until a later stage.

Reply:
Thanks @yujuhong I will update the KEP.

Reply:
@yujuhong I collapsed all the required item in `GA` instead of splitting them between `alpha`, `beta` and `GA`. Let me know if I captured that correctly. Thanks.